src/main/java/org/folio/service/mapping/writer/impl/AbstractRecordWriter.java

```diff
@@ -0,0 +1,105 @@
+package org.folio.service.mapping.writer.impl;
+
+import org.folio.service.mapping.processor.rule.DataSource;
+import org.folio.service.mapping.reader.values.CompositeValue;
+import org.folio.service.mapping.reader.values.ListValue;
+import org.folio.service.mapping.reader.values.SimpleValue;
+import org.folio.service.mapping.reader.values.StringValue;
+import org.folio.service.mapping.writer.RecordWriter;
+import org.folio.service.mapping.writer.fields.RecordControlField;
+import org.folio.service.mapping.writer.fields.RecordDataField;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.folio.service.mapping.reader.values.SimpleValue.SubType.LIST_OF_STRING;
+import static org.folio.service.mapping.reader.values.SimpleValue.SubType.STRING;
+
+public abstract class AbstractRecordWriter implements RecordWriter {
```

Comment:
I don't think we need both abstract class and interface, we can create default methods in the interface and move the method implementations to RecordWriter, so that all the other types can implement just one Interface

Reply:
The purpose of an interface is just to provide a contract to a caller, no need to overload interface with extra logic :)